### ISSUE TITLE
Better matching of 'in' or 'out' users in main.py

### ISSUE THREAD

**Issue by Tarlispace69** [ffamilyfriendly/AuntieRobot#1]
I believe that it would be better to instead of checking for a list of complete statements like "I'm in" You could just check if there is "in" in the sentence and that there's not "lost"/"out" or the opposite. That would make detection more accurate, and I believe that would also make it more efficient as you would have to match fewer characters. 

I'm not familiar with the way you use regex, I usually use `re.findall()`, so I just want to say anyway it's very important for the matching to not be case sensitive. 

You might also want to look into using threading in python to make the program run faster. And next year possibly switching to a lower level more efficient language, at least to handle checking comments and other compute intensive tasks.

good luck cumrade o7.

**Comment by ffamilyfriendly** [comment]
I did originally contemplated something like that but it would lead to a lot of false positives like "im **in** sweden", "when i looked **out** it was snowing", and so forth. People tend to write quite a lot in their answers and both the use "in" and "out" are unfortunately quite common words. 

I dont remember how I did my regex as I never use python but knowing myself I'd make sure the regex was case-insensitive.

The performance "pitfalls" in the code is really only reddits ratelimiting. Checking all data entries for a specific post takes only fractions of a second. I don't particularly see a need to re-write this project as it works and is not really what I'm looking to develop atm (got a backlog of other projects screaming for attention) but if i were to change my mind some day I'd probably write it in rust
